GitCommands/PathUtil.cs

```diff
@@ -332,5 +332,28 @@ string FindFile(string location, string fileName1)
                 return null;
             }
         }
+
+        [ContractAnnotation("=>false,errorMessage:null")]
+        [ContractAnnotation("=>true,errorMessage:notnull")]
+        public static bool TryDeleteDirectory([NotNull] this string path, out string errorMessage)
+        {
+            errorMessage = null;
+            if (string.IsNullOrWhiteSpace(path) || !Directory.Exists(path))
+            {
+                return true;
```

Comment:
This breaches the contract. Here I could agree on failing silently.

Though in general, what is the convention for `TryFoo` functions? I think they should return true on success.

Reply:
I'm not aware of any conventions on the return. Following my logic (debatable :)) if there is nothing to delete - it is a success. 
Alternatively, if we return `false` we may be crashing for no good reason.

Thoughts?

Reply:
You have defined `[ContractAnnotation("=>true,errorMessage:notnull")`, I read this as "return `true` and provide a message on error".
E.g. `TryGet` returns `true` on success, which is more intuitive to me.
I agree return success if the folder does not exist.

Reply:
Ahh, I got it reversed! Thank you